### ISSUE TITLE
Report TextTrack loading/parsing errors as player errors (with new error code)

### ISSUE THREAD

**Issue by seanjohnson08** [videojs/video.js#6754]
## Description
When loading TextTracks of any kind (I'll go with Captions as an example), if these tracks fail to load or fail to parse, we only log errors to the console and we don't fire any callbacks/events.

Here's where that happens for captions: https://github.com/videojs/video.js/blob/0c081a3520b2776756abf5f7ad8a7310ade59995/src/js/tracks/text-track.js#L53

It would be great if we could just add a new player error for text tracks with its own error code. This would enable plugin authors to add tracking or better error handling around tracks in general. For my product use case, having captions is non-negotiable and if they fail to load it should count as a player failure at least. Since I want to add my own logging/tracking around this failure, I'm having to do rather hacky-ish things by inspecting `player.log.history` instead of a global `error` hook.

**Comment by gkatsev** [comment]
Yeah, we should do a better job at surfacing errors and not just for text tracks. Right now errors sort of mean that playback has stopped and cannot continue, so, it's a bit tricky.
